### PR TITLE
fix: typescript sdk: use cursor when accumulating accounts for transfer

### PR DESCRIPTION
**Pull request by sergeytimoshin**
If we have > than 1000 accounts with 0 balance, we need to fetch accounts with cursor.

### CODE REVIEW

**Review by ananas-block — APPROVED** [review]

File: js/stateless.js/src/rpc.ts [hunk 1]
```diff
@@ -664,7 +664,7 @@ export class Rpc extends Connection implements CompressionApiInterface {
      */
     async getCompressedAccountsByOwner(
         owner: PublicKey,
-        config?: GetCompressedAccountsByOwnerConfig,
+        config?: GetCompressedAccountsByOwnerConfig | undefined,
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/rpc.ts:667]: can remove `undefined`, `?` already takes care of it

File: js/stateless.js/src/rpc-interface.ts [hunk 2]
```diff
@@ -504,7 +504,9 @@ export interface CompressionApiInterface {
 
     getCompressedAccountsByOwner(
         owner: PublicKey,
-        config?: GetCompressedAccountsByOwnerConfig,
+        config?:
+            | Omit<GetCompressedAccountsByOwnerConfig, 'cursor' | 'limit'>
+            | undefined,
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/rpc-interface.ts:509]: can remove `undefined`, `?` already takes care of it

why exclude `cursor` and `limit` from the param?

File: js/stateless.js/src/rpc-interface.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ export interface LatestNonVotingSignatures {
 export interface GetCompressedAccountsByOwnerConfig {
     filters?: GetCompressedAccountsFilter[];
     dataSlice?: DataSlice;
-    cursor?: string;
+    cursor: string | null;
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/rpc-interface.ts:43]: why replace `undefined` with `null`? this makes the interface inconsistent

File: js/stateless.js/src/test-helpers/test-rpc/test-rpc.ts [hunk 4]
```diff
@@ -335,9 +335,11 @@ export class TestRpc extends Connection implements CompressionApiInterface {
...
     async getCompressedAccountsByOwner(
         owner: PublicKey,
-        _config?: GetCompressedAccountsByOwnerConfig,
+        config?:
+            | Omit<GetCompressedAccountsByOwnerConfig, 'cursor' | 'limit'>
+            | undefined,
     ): Promise<WithCursor<CompressedAccountWithMerkleContext[]>> {
-        if (_config) {
+        if (config) {
             throw new Error(
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/test-helpers/test-rpc/test-rpc.ts:343]: in which scenario would test-rpc reach >1000 events?

File: js/stateless.js/src/actions/transfer.ts [hunk 5]
```diff
@@ -40,14 +47,45 @@ export async function transfer(
     /// TODO: allow multiple
     merkleTree?: PublicKey,
     confirmOptions?: ConfirmOptions,
+    config?: Omit<GetCompressedAccountsByOwnerConfig, 'cursor' | 'limit'>,
```
Comment (SwenSchaeferjohann) [inline js/stateless.js/src/actions/transfer.ts:50]: what's the reasoning for passing filter params along to the `transfer` action?

**Review by SwenSchaeferjohann — CHANGES_REQUESTED** [review]

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**